File: ignite-nest/src/post/infrastructure/repository/post-query.adapter.ts
import { PostInPort } from '../../domain/port/in/post-in.port';
import { Injectable, NotFoundException } from '@nestjs/common';
import { Post } from '../../entity/post.entity';
import { PageRequestDto } from '../../../common/dto/page-request.dto';
import { PageResponseDto } from '../../../common/dto/page-response.dto';
import { PaginationHelper } from '../../../common/helper/pagination-helper.service';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';

@Injectable()
export class PostQueryAdapter implements PostInPort {
  constructor(
    @InjectRepository(Post)
    private readonly postRepository: Repository<Post>,
    private readonly paginationService: PaginationHelper,
  ) {}

  async getPostByIdOrThrow(postId: number): Promise<Post> {
    const post = await this.postRepository
      .findOneOrFail({ where: { id: postId } })
      .catch(() => {
        throw new NotFoundException(`Post id ${postId} not found`);
      });

    return post;
  }

  async findPostList(request: PageRequestDto): Promise<PageResponseDto<Post>> {
    return this.paginationService.paginate(Post, request);
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/post/application/adapter/PostQueryAdapter.kt
package org.rookedsysc.ignitespring.post.application.adapter

import org.rookedsysc.ignitespring.post.domain.port.`in`.PostInPort
import org.rookedsysc.ignitespring.post.entity.PostEntity
import org.rookedsysc.ignitespring.post.infrastructure.repository.PostRepository
import org.springframework.data.crossstore.ChangeSetPersister.NotFoundException
import org.springframework.stereotype.Repository

@Repository
class PostQueryAdapter(
    private val postRepository: PostRepository
): PostInPort {

    override fun getPostByIdOrThrow(postId: Long): PostEntity {
        return postRepository.findById(postId).orElseThrow { throw NotFoundException() }
    }
}
